src/Money.php

```diff
@@ -56,8 +56,13 @@
      */
     public function __construct($amount, Currency $currency)
     {
-        if (filter_var($amount, FILTER_VALIDATE_INT) === false && !Number::fromString($amount)->isInteger()) {
-            throw new \InvalidArgumentException('Amount must be an integer(ish) value');
+        if (filter_var($amount, FILTER_VALIDATE_INT) === false) {
+            $numberFromString = Number::fromString($amount);
+            if (!$numberFromString->isInteger()) {
+                throw new \InvalidArgumentException('Amount must be an integer(ish) value');
+            }
+
+            $amount = $numberFromString->getIntegerPart();
```

Comment:
Before, if you pass 3.000, then the private property 'amount' would be set to '3.000'. If you then do equals on 'new Money (3, new Currency())' it would fail. Afterwards, it will not. It is a good solution.